src/App.jsx

```diff
@@ -14,33 +15,112 @@ export const goodsFromServer = [
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABET = 'aplha';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPrepearedGoods(goods, sortField, isReversed) {
+  const prepearedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_ALPHABET:
+        prepearedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SORT_FIELD_LENGTH:
+        prepearedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        return prepearedGoods;
+    }
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (isReversed) {
+    return prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const [isResetButtonVisible, setIsResetButtonVisible] = useState(false);
+
+  const resetSort = () => {
+    setSortField('');
+    setIsReversed(false);
+    setIsResetButtonVisible(false);
+  };
+
+  const handleSortClick = field => {
+    getPrepearedGoods(goodsFromServer, field);
+    setIsResetButtonVisible(true);
+    setSortField(field);
+  };
+
+  const handleReverseClick = () => {
+    setIsReversed(prevIsReversed => {
+      const newReversed = !prevIsReversed;
+
+      setIsResetButtonVisible(
+        prevIsResetButtonVisible => sortField || newReversed,
+      );
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+      return newReversed;
+    });
+  };
+
+  const visibleGoods = getPrepearedGoods(
+    goodsFromServer,
+    sortField,
+    isReversed,
+  );
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => handleSortClick(SORT_FIELD_ALPHABET)}
+          className={`button is-info ${sortField === SORT_FIELD_ALPHABET ? '' : 'is-light'}`}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => handleSortClick(SORT_FIELD_LENGTH)}
+          className={`button is-success ${sortField === SORT_FIELD_LENGTH ? '' : 'is-light'}`}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          onClick={handleReverseClick}
+          className={`button is-warning ${isReversed ? '' : 'is-light'}`}
+        >
+          Reverse
+        </button>
+
+        {isResetButtonVisible && (
```

Comment:
```suggestion
        {isResetButtonVisible && (
```
No need to create a state for it. Consider using a variable here